pkg/client/didexchange/client.go

```diff
@@ -101,12 +123,36 @@ func (c *Client) CreateInvitation(label string) (*Invitation, error) {
 		return nil, fmt.Errorf("failed CreateSigningKey: %w", err)
 	}
 
+	var routingKeys []string
+
+	serviceEndpoint := c.inboundTransportEndpoint
+	routerRegistered := true
+
+	routeConf, err := c.routeSvc.Config()
+	if err != nil && !errors.Is(err, route.ErrRouterNotRegistered) {
+		return nil, fmt.Errorf("fetch router config : %w", err)
+	} else if errors.Is(err, route.ErrRouterNotRegistered) {
+		routerRegistered = false
+	}
+
+	if routeConf != nil {
+		serviceEndpoint = routeConf.Endpoint()
+		routingKeys = routeConf.Keys()
+	}
+
 	invitation := &didexchange.Invitation{
 		ID:              uuid.New().String(),
 		Label:           label,
 		RecipientKeys:   []string{sigPubKey},
-		ServiceEndpoint: c.inboundTransportEndpoint,
+		ServiceEndpoint: serviceEndpoint,
 		Type:            didexchange.InvitationMsgType,
+		RoutingKeys:     routingKeys,
+	}
+
+	if routerRegistered {
+		if err = c.routeSvc.AddKey(sigPubKey); err != nil && !errors.Is(err, route.ErrRouterNotRegistered) {
+			return nil, fmt.Errorf("create invitation - add key to the router : %w", err)
+		}
```

Comment:
this one too

Reply:
[Moved](https://github.com/hyperledger/aries-framework-go/compare/3833ed3479ac417972cff8aad5a07230c2f6f54b..9dcd5f3aff2537b1907bafee711df2921a81201e#diff-427522b353a28b1ec350c29c79d387b0R132) to a helper function.